task/upload.go

```diff
@@ -395,14 +403,16 @@ func outputLocations(outURL string, relativePath string) ([]OutputName, []client
 	if err != nil {
 		return nil, nil, fmt.Errorf("error parsing object store URL: %w", err)
 	}
+	// source segments cannot be stored in web3.storage
+	sourceSegments := !strings.HasPrefix(outURL, "w3s")
```

Comment:
Neat logic! A couple of questions:
 - Are we not storing HLS playlists in web3.storage at all, or only not the source segments rendition?
 - Does `catalyst-api` handle these settings correctly? I remember it got lost in its core pipeline, not sure if it's changed now.

Reply:
> * Are we not storing HLS playlists in web3.storage at all, or only not the source segments rendition?

We **do** store HLS playlist in manifest files, but **not** the source segments

> * Does `catalyst-api` handle these settings correctly? I remember it got lost in its core pipeline, not sure if it's changed now.

Yes, `catalyst-api` handles that correctly.